compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -382,22 +382,23 @@ updateFileDiagnostics ::
   -> Maybe [FileDiagnostic] -- ^ previous results for this file
   -> [FileDiagnostic] -- ^ current results
   -> Action ()
-updateFileDiagnostics afp previousAll currentAll = do
+updateFileDiagnostics fp previousAll currentAll = do
     -- TODO (MK) We canonicalize to make sure that the two files agree on use of
     -- / and \ and other shenanigans.
     -- Once we have finished the migration to haskell-lsp we should make sure that
     -- this is no longer necessary.
-    afp' <- liftIO $ canonicalizePath afp
+    afp <- liftIO $ canonicalizePath fp
+    let uri = filePathToUri afp
     let filtM diags = do
             diags' <-
                 filterM
-                    (\x -> fmap (== afp') (canonicalizePath $ fst x))
+                    (\x -> fmap (== afp) (canonicalizePath $ fst x))
```

Comment:
I think this code is dodgy as hell. It was certainly broken earlier today, and I have no confidence it won't break again, or even isn't broken now in some circumstance I can't imagine. I think it just wants to die, or become obviously correct.